src/vm/runners/builtin_runner/hash.rs

```diff
@@ -12,22 +13,24 @@ use crate::vm::vm_memory::memory_segments::MemorySegmentManager;
 
 pub struct HashBuiltinRunner {
     pub base: isize,
+    stop_ptr: Option<Relocatable>,
+    name: String,
     _ratio: usize,
     cells_per_instance: usize,
     _n_input_cells: usize,
-    _stop_ptr: Option<Relocatable>,
     verified_addresses: Vec<Relocatable>,
 }
 
 impl HashBuiltinRunner {
-    pub fn new(ratio: usize) -> Self {
+    pub fn new(name: impl Into<String>, ratio: usize) -> Self {
```

Comment:
This extra field is not needed, the name of the builtin is fixed and doesn't depend on external factors, we can make it into a constant (defined in each builtin runner's file) if its more convenient.

Reply:
In the reference implementation, it [could also implement](https://github.com/starkware-libs/cairo-lang/blob/54d7e92a703b3b5a1e07e9389608178129946efc/src/starkware/starknet/core/os/class_hash.py#L83) a `"custom_hasher"`. That's why it's a field right now.

Should I disregard other possible implementations?

Reply:
I guess not right now since it seems to be hardcoded to `petersen` in `cairo-rs` for now.

Reply:
Yes, we currently don't support custom hashers in the HashBuiltinRunner